main.cpp

```diff
@@ -1,23 +1,197 @@
 #include <iostream>
 #include <string>
 #include <string_view>
+#include <sstream>
+#include <vector>
 #include <algorithm>
 
 namespace
 {
+int random_no(unsigned int range)
+{
```

Comment:
Please use the `random` header.

```cpp
s32 random_number(s32 min, s32 max)
{
    static std::random_device rd;
    static std::default_random_engine engine(rd());
    std::uniform_int_distribution<s32> dist(min, max);
    return dist(engine);
}
```

Reply:
`random()` isn't allowed?

Reply:
More like `rand()` is deprecated; it is unreliable, non-uniform, and should not be used in modern C++ at all.